receiver/awscontainerinsightreceiver/internal/gpu/dcgmscraper.go

```diff
@@ -0,0 +1,235 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gpu
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+)
+
+const (
+	caFile             = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	collectionInterval = 60 * time.Second
+	jobName            = "containerInsightsDCGMExporterScraper"
+)
+
+type DcgmScraper struct {
+	ctx                context.Context
+	settings           component.TelemetrySettings
+	host               component.Host
+	hostInfoProvider   hostInfoProvider
+	prometheusReceiver receiver.Metrics
+	running            bool
+}
+
+type DcgmScraperOpts struct {
+	Ctx               context.Context
+	TelemetrySettings component.TelemetrySettings
+	Consumer          consumer.Metrics
+	Host              component.Host
+	HostInfoProvider  hostInfoProvider
+	BearerToken       string
+}
+
+type hostInfoProvider interface {
+	GetClusterName() string
+	GetInstanceID() string
+}
+
+func NewDcgmScraper(opts DcgmScraperOpts) (*DcgmScraper, error) {
+	if opts.Consumer == nil {
+		return nil, errors.New("consumer cannot be nil")
+	}
+	if opts.Host == nil {
+		return nil, errors.New("host cannot be nil")
+	}
+	if opts.HostInfoProvider == nil {
+		return nil, errors.New("cluster name provider cannot be nil")
+	}
+
+	scrapeConfig := &config.ScrapeConfig{
+		// TLS needs to be enabled between pods communication
+		// It can further get restricted by adding authentication mechanism to limit the data
+		//HTTPClientConfig: configutil.HTTPClientConfig{
+		//	BasicAuth: &configutil.BasicAuth{
+		//		Username: "",
+		//		Password: "",
+		//	},
+		//	Authorization: &configutil.Authorization{
+		//		Type: "basic_auth",
+		//	},
+		//	TLSConfig: configutil.TLSConfig{
+		//		CAFile:             caFile,
+		//		InsecureSkipVerify: false,
+		//	},
+		//},
```

Comment:
Is this meant to be commented out?

Reply:
yes, until we have TLS work is done in the operator

Reply:
nit: Would prefer if we didn't include this until the TLS work is done.